helper/pluginutil/runner.go

```diff
@@ -0,0 +1,109 @@
+package pluginutil
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/helper/mlock"
+)
+
+var (
+	// PluginUnwrapTokenEnv is the ENV name used to pass unwrap tokens to the
+	// plugin.
+	PluginMlockEnabled = "VAULT_PLUGIN_MLOCK_ENABLED"
+)
+
+// Looker defines the plugin Lookup function that looks into the plugin catalog
+// for availible plugins and returns a PluginRunner
+type Looker interface {
+	LookupPlugin(string) (*PluginRunner, error)
+}
+
+// Wrapper interface defines the functions needed by the runner to wrap the
+// metadata needed to run a plugin process. This includes looking up Mlock
+// configuration and wrapping data in a respose wrapped token.
+type Wrapper interface {
+	ResponseWrapData(data map[string]interface{}, ttl time.Duration, jwt bool) (string, error)
+	MlockDisabled() bool
+}
+
+// LookWrapper defines the functions for both Looker and Wrapper
+type LookWrapper interface {
+	Looker
+	Wrapper
+}
+
+// PluginRunner defines the metadata needed to run a plugin securely with
+// go-plugin.
+type PluginRunner struct {
+	Name           string                      `json:"name"`
+	Command        string                      `json:"command"`
+	Args           []string                    `json:"args"`
+	Sha256         []byte                      `json:"sha256"`
+	Builtin        bool                        `json:"builtin"`
+	BuiltinFactory func() (interface{}, error) `json:"-"`
+}
+
+// Run takes a wrapper instance, and the go-plugin paramaters and executes a
+// plugin.
+func (r *PluginRunner) Run(wrapper Wrapper, pluginMap map[string]plugin.Plugin, hs plugin.HandshakeConfig, env []string) (*plugin.Client, error) {
+	// Get a CA TLS Certificate
+	certBytes, key, err := GenerateCert()
+	if err != nil {
+		return nil, err
+	}
+
+	// Use CA to sign a client cert and return a configured TLS config
+	clientTLSConfig, err := CreateClientTLSConfig(certBytes, key)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use CA to sign a server cert and wrap the values in a response wrapped
+	// token.
+	wrapToken, err := WrapServerConfig(wrapper, certBytes, key)
+	if err != nil {
+		return nil, err
+	}
+
+	mlock := "true"
+	if wrapper.MlockDisabled() {
```

Comment:
I would suggest inverting the logic of MlockDisabled. Then you can just `mlock := wrapper.MlockEnabled()` here and anywhere else something may call it. I realize that it's the opposite of the config directive, but the code is a little nicer.